### ISSUE TITLE
Continue with the question of DDNS automatic stop

### ISSUE THREAD

**Issue by livepu** [ericpaulbishop/gargoyle#831]
Through its own system records, it is found that https://aruljohn.com/, the acquired IP is sometimes incorrect. I actually got the IP address 10.20.10.22.

https://github.com/ericpaulbishop/gargoyle/blob/master/package/ddns-gargoyle/src/check_local_ip_urls.c
I find that here, logically, IP is always acquired through the network. Instead of local interface to get IP

**Comment by lantis1008** [comment]
It is already solved by https://github.com/lantis1008/gargoyle/commit/407a3b42eaca9945518d6848c7a91a27410fac58
But has not been merged back into Master yet.

**Comment by livepu** [comment]
Get IP through local interface to judge whether it is reasonable or not. If not, would it be better to get it online?

**Comment by lantis1008** [comment]
It supports fetching from a local interface or online, but not both.
We rely on external services to have their systems working fine, if not we move away from them. I tried to contact @aruljohn on Facebook and he never replied. 

Perhaps he doesn't want to allow his service to be pinged by routers on a semi regular basis, or there is something else going on.

**Comment by livepu** [comment]
Is it better to use local interface to judge ip?
In addition, when to release 19.07

**Comment by lantis1008** [comment]
That depends on your particular setup. If you have a double NAT type scenario or router that is not directly internet facing, then no. 

I don't know yet, OpenWrt haven't even released it upstream. I'm just doing preparatory development works.

**Comment by livepu** [comment]
Can add a manual selection to the ddns to identify the ip method? I think this is closer to the actual needs.

**Comment by lantis1008** [comment]
You can already set it in the config file. 
I don't think this is a use case for most users, so I don't want to confuse the issue by adding it to the GUI

**Comment by livepu** [comment]
I edit  /etc/ddns_gargoyle.conf，like this
`service ddns_1
        enabled 1
        service_provider        3322.org
        ip_source       INTERFACE
        force_interval  1
        force_unit      days
        check_interval  10
        check_unit      minutes`
but system log show error
`Mon Sep  2 13:14:21 2019 daemon.notice procd: /etc/rc.d/S90ddns_gargoyle: ERROR: No valid dynamic DNS service configurations defined
Mon Sep  2 13:14:21 2019 daemon.notice procd: /etc/rc.d/S90ddns_gargoyle: (Did you specify correct configuration file path?)
Mon Sep  2 13:14:21 2019 daemon.notice netifd: Network device 'wlan0' link is up`

**Comment by lantis1008** [comment]
Need ip_interface as well. 
I assume this could be "wan", but may need to be "eth0" etc.

**Comment by livepu** [comment]
I try 3 times . only like this can success! thank you!
```
service ddns_1
        enabled 1
        service_provider        3322.org
        ip_source       INTERFACE
        ip_interface    pppoe-wan
```

**Comment by lantis1008** [comment]
I'm glad you are sorted. 

As I was unable to reproduce your original issue, can you please close this ticket?
If other users have a problem in the future, we can reopen it. 
Thanks.

**Comment by EricFromCanada** [comment]
I've also just run into this. I have a TL-WDR4300 v1.3 running Gargoyle 1.12.0 in a double NAT setup, and every so often it incorrectly updates my domain name on ChangeIP to either 10.20.10.22 or 127.0.0.1 (which may be [caused by Adblock](https://www.gargoyle-router.com/phpbb/viewtopic.php?t=7367); I have v20191111-1.2.1 installed). 

<img width="1041" alt="Screen Shot" src="https://user-images.githubusercontent.com/3324775/95507952-3b487400-0980-11eb-843a-b1c0c8dc9cd7.png">

**Comment by EricFromCanada** [comment]
I've found the problematic results using `logread`: 
```
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: Checking whether update needed:
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: 	service provider=changeip.com
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: 	domain=x.x.info
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: 		Successfully retrieved local ip from url: https://api.my-ip.io/ip
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: 	local IP  = x.x.x.147
Thu Oct  8 14:16:40 2020 daemon.info ddns_gargoyle[2214]: 	remote IP = 127.0.0.1
Thu Oct  8 14:16:41 2020 daemon.info ddns_gargoyle[2214]: 	Update successful
```

```
Thu Oct  8 16:11:53 2020 daemon.info ddns_gargoyle[4911]: Checking whether update needed:
Thu Oct  8 16:11:53 2020 daemon.info ddns_gargoyle[4911]: 	service provider=dnsomatic.com
Thu Oct  8 16:11:53 2020 daemon.info ddns_gargoyle[4911]: 		Successfully retrieved local ip from url: http://www.lawrencegoetz.com/programs/ipinfo/
Thu Oct  8 16:11:53 2020 daemon.info ddns_gargoyle[4911]: 	local IP  = x.x.x.147
Thu Oct  8 16:11:53 2020 daemon.info ddns_gargoyle[4911]: 	remote IP = 10.20.10.22
Thu Oct  8 16:11:54 2020 daemon.info ddns_gargoyle[4911]: 	Update successful
```

**Comment by EricFromCanada** [comment]
I was able to route around the issue by mapping the uncooperative servers to one of the IANA's [non-routable addresses](https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml) in the router's `/etc/hosts`.
```
203.0.113.1    aruljohn.com
203.0.113.1    bot.whatismyipaddress.com
```

**Comment by supermarkert** [comment]
FYI, I'm also having this same exact problem on latest Gargoyle release 1.12 (my IP resolves as 10.20.10.22 sometimes). But I'm somewhat of a noob here, so is there a simple suggested workaround? I've tried following several threads about this, and there doesn't seem to be a unilateral suggestion.

**Comment by EricFromCanada** [comment]
I wrote a simple workaround in my [last comment](https://github.com/ericpaulbishop/gargoyle/issues/831#issuecomment-709321301).

**Comment by lantis1008** [comment]
The aruljohn issue is solved in 1.13 and the latest version of 1.12.x.
I will look into the other one, thanks for reporting it.

**Comment by livepu** [comment]
Access to IP through the Internet will be affected by the network environment. Later, I got the IP through the local interface, and there was no error.

https://github.com/ericpaulbishop/gargoyle/pull/917
